src/integration/resources/types.go

```diff
@@ -82,6 +84,25 @@ type Admin interface {
 	Close() error
 }
 
+// M3msgTopicManager is a wrapper for m3msg functions.
+type M3msgTopicManager interface {
```

Comment:
These separate interfaces are unnecessary. Let's move these methods under the `Admin` interface since the coordinator will register all these routes at the same time, if allowed.